app/models/version.rb

```diff
@@ -26,4 +34,22 @@ def to_s
   def self.past_month_activities(objects, donor_id)
     related_to_multiple(objects).except_user(donor_id).past_month
   end
+
+  # required by PushUpdates and PaperTrail modules
+  def offer
+    related_type == "Offer" ? related : nil
+  end
+
+  def valid_pusher_update?
+    item_log? || item_package_log?
+  end
+
+  def item_log?
+    self.item_type == "Item" &&
+    (self.object_changes.keys & %w(donor_description donor_condition_id state)).present?
+  end
+
+  def item_package_log?
+    self.item_type == "Package" && %w(received missing).include?(self.object_changes["state"][1])
+  end
```

Comment:
Thanks @steveyken, Sure, I will change it.
But what will happen with existing data and what if we need some other data in future?